extensions/keycloak-admin-client-common/runtime/src/main/java/io/quarkus/keycloak/admin/client/common/KeycloakAdminClientConfig.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.keycloak.admin.client.common;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Keycloak Admin Client
+ */
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, name = "keycloak.admin-client")
+public class KeycloakAdminClientConfig {
+
+    /**
+     * Realm.
+     */
+    @ConfigItem(defaultValue = "master")
+    public String realm;
+
+    /**
+     * Keycloak server URL, for example, `https://host:port`.
+     */
+    @ConfigItem
+    public Optional<String> serverUrl;
+
+    /**
+     * Client id.
+     */
+    @ConfigItem
+    public Optional<String> clientId;
+
+    /**
+     * Username. Required with a `password` grant type.
+     */
+    @ConfigItem
+    public Optional<String> username;
```

Comment:
I'll leave it up to you, but since `admin:admin` is used a lot in test/demo instances of KC, incl the one launched by DevServices for KC, I'd have no problems with defaulting the username and password to `admin`.

So in a typical case where a user wants to use it to setup a test KC instance, one would only have to set the `server-url`

Reply:
I tried to avoid "necessity" to remove password/username with client_credentials:
```
quarkus.keycloak.admin-client.username=
quarkus.keycloak.admin-client.password=
```
though implementation don't set them with client_credentials anyway https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/integration/admin-client/src/main/java/org/keycloak/admin/client/token/TokenManager.java#L79, it feels wrong to rely on impl.

It looks better with defaults, so I've added them. Thank you